recipes/newrelic/infrastructure/cloud/aws-linux.yml

```diff
@@ -20,10 +20,104 @@ processMatch: []
 preInstall:
   requireAtDiscovery: |
       code=$(curl -I http://169.254.169.254/latest/meta-data/instance-id -w %{response_code} -so '/dev/null')
-      if [ $code == "200" ]; then
-        exit 132
+
+      if [ $code != "200" ]; then
+        exit 1
+      fi
+
+      if [ ! -f ~/.aws/credentials ]; then
+        exit 1
```

Comment:
@noahmmcgivern 
After we know that this is a linux instance running on AWS (curl to IP succeeded), all the exit codes for error should be 132 to indicate a DETECTED status, so the user can see our documentation for manually installing.